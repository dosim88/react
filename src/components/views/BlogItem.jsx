import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';

import Image from 'components/ui/Image';
import TextBox from 'components/ui/TextBox';
import Meta from 'components/ui/Meta';
import LikeContainer from 'components/containers/LikeContainer';

import { posts, postEdit } from 'helpers/routes';

const BlogItem = ({ post }) => (
  <div className='blog-item text-center'>
    <div className="row">
      <TextBox>
        <Link to={posts(post.id)}>{post.title}</Link>
      </TextBox>
    </div>

    <div className="row">
      <div className="col-md-6">
        <Image {...post.image} />
      </div>

      <div className="col-md-6">
        <Meta {...post.meta} />
      </div>
    </div>

    <div className="row">
      <LikeContainer postId={post.id} />
      <Link to={postEdit(post.id)}>Edit</Link>
    </div>
  </div>
);

BlogItem.propTypes = {
  post: PropTypes.shape({
    image:  PropTypes.shape(Image.propTypes),
    meta: PropTypes.shape(Meta.propTypes),
    title: PropTypes.string,
    id: PropTypes.number
  })
};

BlogItem.defaultProps = {
  post: {}
};

export default BlogItem;
