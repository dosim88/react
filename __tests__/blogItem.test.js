import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import { Link } from 'react-router-dom';

import { Provider } from 'react-redux';
import { ConnectedRouter } from 'react-router-redux';

import createBrowserHistory from 'history/createBrowserHistory';
const history = createBrowserHistory();

import fakeStore from 'helpers/tests/fakeStore';

import BlogItem from 'components/views/BlogItem';


describe('BlogItem', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={fakeStore}>
        <ConnectedRouter history={history}>
          <BlogItem />
        </ConnectedRouter>
      </Provider>,
      div
    );
  });

  describe('render', () => {
    it('should render the title', () => {
      const itemProps = { title: 'Hello, World!', id: 5 };

      const item = shallow(<BlogItem post={itemProps} />);

      const header = <Link to="/posts/5">Hello, World!</Link>;

      expect(item.contains(header)).toEqual(true);
    });
  });


  it('should render usual item', () => {
    const itemProps = {
      title: 'Hello, World!',
      id: 5,
      image: {
        src: 'http://localhost:3001/images/post-1.jpg',
        alt: 'Hello World!'
      }
    };

    const item = shallow(<BlogItem post={itemProps} />);

    expect(item).toMatchSnapshot();
  });
});

