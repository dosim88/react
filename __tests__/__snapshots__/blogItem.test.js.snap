// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="blog-item text-center"
>
    <div
        className="row"
    >
        <TextBox>
            <Link
                replace={false}
                to="/posts/5"
            >
                Hello, World!
            </Link>
        </TextBox>
    </div>
    <div
        className="row"
    >
        <div
            className="col-md-6"
        >
            <Image
                alt="Hello World!"
                src="http://localhost:3001/images/post-1.jpg"
            />
        </div>
        <div
            className="col-md-6"
        >
            <Meta
                author="нет данных"
                createdAt="нет данных"
                likesCount={0}
                updatedAt="нет данных"
            />
        </div>
    </div>
    <div
        className="row"
    >
        <Connect(Like)
            postId={5}
        />
        <Link
            replace={false}
            to="/posts/5/edit"
        >
            Edit
        </Link>
    </div>
</div>,
  "nodes": Array [
    <div
      className="blog-item text-center"
>
      <div
            className="row"
      >
            <TextBox>
                  <Link
                        replace={false}
                        to="/posts/5"
                  >
                        Hello, World!
                  </Link>
            </TextBox>
      </div>
      <div
            className="row"
      >
            <div
                  className="col-md-6"
            >
                  <Image
                        alt="Hello World!"
                        src="http://localhost:3001/images/post-1.jpg"
                  />
            </div>
            <div
                  className="col-md-6"
            >
                  <Meta
                        author="нет данных"
                        createdAt="нет данных"
                        likesCount={0}
                        updatedAt="нет данных"
                  />
            </div>
      </div>
      <div
            className="row"
      >
            <Connect(Like)
                  postId={5}
            />
            <Link
                  replace={false}
                  to="/posts/5/edit"
            >
                  Edit
            </Link>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        post={
                Object {
                        "id": 5,
                        "image": Object {
                          "alt": "Hello World!",
                          "src": "http://localhost:3001/images/post-1.jpg",
                        },
                        "title": "Hello, World!",
                      }
        }
/>,
      "_debugID": 32,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "post": Object {
            "id": 5,
            "image": Object {
              "alt": "Hello World!",
              "src": "http://localhost:3001/images/post-1.jpg",
            },
            "title": "Hello, World!",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="blog-item text-center"
>
          <div
                    className="row"
          >
                    <TextBox>
                              <Link
                                        replace={false}
                                        to="/posts/5"
                              >
                                        Hello, World!
                              </Link>
                    </TextBox>
          </div>
          <div
                    className="row"
          >
                    <div
                              className="col-md-6"
                    >
                              <Image
                                        alt="Hello World!"
                                        src="http://localhost:3001/images/post-1.jpg"
                              />
                    </div>
                    <div
                              className="col-md-6"
                    >
                              <Meta
                                        author="нет данных"
                                        createdAt="нет данных"
                                        likesCount={0}
                                        updatedAt="нет данных"
                              />
                    </div>
          </div>
          <div
                    className="row"
          >
                    <Connect(Like)
                              postId={5}
                    />
                    <Link
                              replace={false}
                              to="/posts/5/edit"
                    >
                              Edit
                    </Link>
          </div>
</div>,
        "_debugID": 33,
        "_renderedOutput": <div
          className="blog-item text-center"
>
          <div
                    className="row"
          >
                    <TextBox>
                              <Link
                                        replace={false}
                                        to="/posts/5"
                              >
                                        Hello, World!
                              </Link>
                    </TextBox>
          </div>
          <div
                    className="row"
          >
                    <div
                              className="col-md-6"
                    >
                              <Image
                                        alt="Hello World!"
                                        src="http://localhost:3001/images/post-1.jpg"
                              />
                    </div>
                    <div
                              className="col-md-6"
                    >
                              <Meta
                                        author="нет данных"
                                        createdAt="нет данных"
                                        likesCount={0}
                                        updatedAt="нет данных"
                              />
                    </div>
          </div>
          <div
                    className="row"
          >
                    <Connect(Like)
                              postId={5}
                    />
                    <Link
                              replace={false}
                              to="/posts/5/edit"
                    >
                              Edit
                    </Link>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    post={
        Object {
            "id": 5,
            "image": Object {
              "alt": "Hello World!",
              "src": "http://localhost:3001/images/post-1.jpg",
            },
            "title": "Hello, World!",
          }
    }
/>,
}
`;
